Does not support `gitlab:` urls

### pnpm version: 6.11.5
On Linux (debian testing), node 14.17.4

npm supports these `gitlab:` urls, but pnpm does not seem to.

```
$ pnpm i gitlab:Eranziel/foundry-pc-types
 WARN  GET https://gitlab.com/Eranziel/foundry-pc-types/repository/archive.tar.gz?ref=9b2b29b507bae55a6da869fc6ea6f831b4e423dd error (ERR_PNPM_FETCH_503). Will retry in 10 seconds. 2 retries left.
^C
$ npm i gitlab:Eranziel/foundry-pc-types 
...

+ foundry-pc-types@0.1.0
added 13 packages from 85 contributors and audited 13 packages in 6.159s
found 0 vulnerabilities
```

Thanks!

am getting the same issue when using gitlab urls in my package.json files, works fine with npm / yarn

```
 WARN  GET https://gitlab.com/coboxcoop/crypto-encoder/repository/archive.tar.gz?ref=e67b0d640ad55234ae8716d7a52569a61c208691 error (ERR_PNPM_FETCH_503). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://gitlab.com/coboxcoop/hyperdrive/repository/archive.tar.gz?ref=bba9eecaf73cb138c179e24b978bbc9db46538cb error (ERR_PNPM_FETCH_503). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://gitlab.com/coboxcoop/hyperdrive/repository/archive.tar.gz?ref=b20f935efd39a51ea1dc2a0368c0608c1bf169a0 error (ERR_PNPM_FETCH_503). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://gitlab.com/coboxcoop/kappa-drive-mount/repository/archive.tar.gz?ref=c2fe12055c8f9106b77a6ee61c8003c089cb0cb7 error (ERR_PNPM_FETCH_503). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://gitlab.com/coboxcoop/crypto-encoder/repository/archive.tar.gz?ref=e67b0d640ad55234ae8716d7a52569a61c208691 error (ERR_PNPM_FETCH_503). Will retry in 1 minute. 1 retries left.
 WARN  GET https://gitlab.com/coboxcoop/hyperdrive/repository/archive.tar.gz?ref=bba9eecaf73cb138c179e24b978bbc9db46538cb error (ERR_PNPM_FETCH_503). Will retry in 1 minute. 1 retries left.
 WARN  GET https://gitlab.com/coboxcoop/hyperdrive/repository/archive.tar.gz?ref=b20f935efd39a51ea1dc2a0368c0608c1bf169a0 error (ERR_PNPM_FETCH_503). Will retry in 1 minute. 1 retries left.
 WARN  GET https://gitlab.com/coboxcoop/kappa-drive-mount/repository/archive.tar.gz?ref=c2fe12055c8f9106b77a6ee61c8003c089cb0cb7 error (ERR_PNPM_FETCH_503). Will retry in 1 minute. 1 retries left.
```

--- a/package.json
+++ b/package.json
@@ -60,6 +60,7 @@
       "@yarnpkg/parsers@2": "2.3.0",
       "glob-parent@3": "^5.1.2",
       "hosted-git-info@<3.0.8": "^3.0.8",
+      "hosted-git-info@4": "npm:@zkochan/hosted-git-info@^4.0.2",
       "http-errors": "^1.7.3",
       "istanbul-reports": "npm:@zkochan/istanbul-reports",
       "js-yaml@^4.0.0": "npm:@zkochan/js-yaml@0.0.5",

--- a/packages/git-resolver/package.json
+++ b/packages/git-resolver/package.json
@@ -34,7 +34,7 @@
     "@pnpm/fetch": "workspace:4.0.2",
     "@pnpm/resolver-base": "workspace:8.0.4",
     "graceful-git": "^3.0.2",
-    "hosted-git-info": "^4.0.1",
+    "hosted-git-info": "npm:@zkochan/hosted-git-info@^4.0.2",
     "semver": "^7.3.4"
   },
   "devDependencies": {

--- a/packages/git-resolver/test/index.ts
+++ b/packages/git-resolver/test/index.ts
@@ -265,7 +265,7 @@ test('resolveFromGit() gitlab with commit', async () => {
     id: 'gitlab.com/pnpm/git-resolver/988c61e11dc8d9ca0b5580cb15291951812549dc',
     normalizedPref: 'gitlab:pnpm/git-resolver#988c61e11dc8d9ca0b5580cb15291951812549dc',
     resolution: {
-      tarball: 'https://gitlab.com/pnpm/git-resolver/repository/archive.tar.gz?ref=988c61e11dc8d9ca0b5580cb15291951812549dc',
+      tarball: 'https://gitlab.com/api/v4/projects/pnpm%2Fgit-resolver/repository/archive.tar.gz?ref=988c61e11dc8d9ca0b5580cb15291951812549dc',
     },
     resolvedVia: 'git-repository',
   })

--- a/packages/git-resolver/test/index.ts
+++ b/packages/git-resolver/test/index.ts
@@ -279,7 +279,7 @@ test('resolveFromGit() gitlab with no commit', async () => {
     id: `gitlab.com/pnpm/git-resolver/${hash}`,
     normalizedPref: 'gitlab:pnpm/git-resolver',
     resolution: {
-      tarball: `https://gitlab.com/pnpm/git-resolver/repository/archive.tar.gz?ref=${hash}`,
+      tarball: `https://gitlab.com/api/v4/projects/pnpm%2Fgit-resolver/repository/archive.tar.gz?ref=${hash}`,
     },
     resolvedVia: 'git-repository',
   })
@@ -293,7 +293,7 @@ test('resolveFromGit() gitlab with branch', async () => {
     id: `gitlab.com/pnpm/git-resolver/${hash}`,
     normalizedPref: 'gitlab:pnpm/git-resolver#master',
     resolution: {
-      tarball: `https://gitlab.com/pnpm/git-resolver/repository/archive.tar.gz?ref=${hash}`,
+      tarball: `https://gitlab.com/api/v4/projects/pnpm%2Fgit-resolver/repository/archive.tar.gz?ref=${hash}`,
     },
     resolvedVia: 'git-repository',
   })
@@ -305,7 +305,7 @@ test('resolveFromGit() gitlab with tag', async () => {
     id: 'gitlab.com/pnpm/git-resolver/87cf6a67064d2ce56e8cd20624769a5512b83ff9',
     normalizedPref: 'gitlab:pnpm/git-resolver#0.3.4',
     resolution: {
-      tarball: 'https://gitlab.com/pnpm/git-resolver/repository/archive.tar.gz?ref=87cf6a67064d2ce56e8cd20624769a5512b83ff9',
+      tarball: 'https://gitlab.com/api/v4/projects/pnpm%2Fgit-resolver/repository/archive.tar.gz?ref=87cf6a67064d2ce56e8cd20624769a5512b83ff9',
     },
     resolvedVia: 'git-repository',
   })

